### PR TITLE
[TEP-0129] Multiple Tekton instances per Cluster

### PR DESCRIPTION
**Pull request by afrittoli**
TEP proposal about running multiple instances of Tekton Pipeline in a single Tekton cluster.

Signed-off-by: Andrea Frittoli <andrea.frittoli@gmail.com>

### CODE REVIEW

**Comment by pritidesai** [comment]
Thank you @afrittoli for the proposal 👍 

Here is the reference to the feature request opened by @aiden-deloryn - https://github.com/tektoncd/pipeline/issues/4605 Thank you @aiden-deloryn 🙏

**Comment by rafalbigaj** [comment]
Thank you @afrittoli for collecting all requirements. This is highly awaited features 👍

**Comment by afrittoli** [comment]
Thank you @pritidesai - I added the reference to the issue.
I moved this to TEP-0129 as TEP-0128 is already taken in another PR.

File: teps/0129-multiple-tekton-instances-per-cluster.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+  as long as there are no backward incompatible changes between the older and newer
+  versions.
+- Identify and configure cluster level resources in Tekton Pipelines, such as CRDs,
+  cluster roles and bindings and the conversion webhook, so that they can safely
+  shared across multiple Tekton instances.
+
+### Non-Goals
+
+- Support multiple instances of Tekton Triggers
+- Changes to the Tekton cli, dashboard, operator, results, chains, triggers to
```
Comment (afrittoli) [inline teps/0129-multiple-tekton-instances-per-cluster.md:77]: We may want to include the operator here, to make sure that uninstalling an operator-installed Tekton does not break other instances of Tekton running in the same cluster. 
Alternatively we could exclude operator-installed tekton in the first iteration and add operator support as a follow-up

**Review by dibyom — APPROVED** [review]

File: teps/0129-multiple-tekton-instances-per-cluster.md [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+cluster.
+
+## Motivation
+
+The Tekton controller provides some functionality towards the ability of
+running multiple instances in a single cluster, namely the ability to
+configure the Tekton controller to watch a single namespace for resources
+to be reconciled. This functionality is not properly documented and it does
+not include other Tekton controllers, such as the various controllers
+bundled in the Tekton webhook as well the remote resolution controller.
```
Comment (pritidesai) [inline teps/0129-multiple-tekton-instances-per-cluster.md:59]: Is it by choice or something back then prevented us to implement `webhook` controller to pipeline controller at the namespace level?

**Review by pritidesai — APPROVED** [review]
thank you @afrittoli for this proposal 👍

**Comment by pritidesai** [comment]
@afrittoli please rebase this PR and I am happy to merge it, thanks!

**Comment by dibyom** [comment]
API WG:

/lgtm

**Pull request merged**